### ISSUE TITLE
iOS compilation error: firebase_*/ios/Classes/UserAgent.h: No such file or directory

### ISSUE THREAD

**Issue by momoDragon** [flutter/flutter#42382]
Version: 
`google_maps_flutter 0.5.21+7 `

Steps tried:
`pod install`
`pod update`
`flutter pub cache repair`

## Error given:
```
 ** BUILD FAILED **


Xcode's output:
↳
    In file included from /Users/mozesong/Developer/flutter/.pub-cache/hosted/pub.dartlang.org/google_maps_flutter-0.5.21+7/ios/Classes/GoogleMapsPlugin.m:5:
    In file included from /Users/mozesong/Developer/flutter/.pub-cache/hosted/pub.dartlang.org/google_maps_flutter-0.5.21+7/ios/Classes/GoogleMapsPlugin.h:8:
    /Users/mozesong/Developer/flutter/.pub-cache/hosted/pub.dartlang.org/google_maps_flutter-0.5.21+7/ios/Classes/GoogleMapController.h:14:52: warning: pointer is missing a nullability type specifier (_Nonnull, _Nullable, or _Null_unspecified) [-Wnullability-completeness]
    - (void)setCameraTargetBounds:(GMSCoordinateBounds *)bounds;
                                                       ^
    /Users/mozesong/Developer/flutter/.pub-cache/hosted/pub.dartlang.org/google_maps_flutter-0.5.21+7/ios/Classes/GoogleMapController.h:14:52: note: insert '_Nullable' if the pointer may be null
    - (void)setCameraTargetBounds:(GMSCoordinateBounds *)bounds;
                                                       ^
                                                        _Nullable
    /Users/mozesong/Developer/flutter/.pub-cache/hosted/pub.dartlang.org/google_maps_flutter-0.5.21+7/ios/Classes/GoogleMapController.h:14:52: note: insert '_Nonnull' if the pointer should never be null
    - (void)setCameraTargetBounds:(GMSCoordinateBounds *)bounds;
                                                       ^
                                                        _Nonnull
    /Users/mozesong/Developer/flutter/.pub-cache/hosted/pub.dartlang.org/google_maps_flutter-0.5.21+7/ios/Classes/GoogleMapController.h:37:65: warning: pointer is missing a nullability type specifier (_Nonnull, _Nullable, or _Null_unspecified) [-Wnullability-completeness]
                        registrar:(NSObject<FlutterPluginRegistrar> *)registrar;
                                                                    ^
    /Users/mozesong/Developer/flutter/.pub-cache/hosted/pub.dartlang.org/google_maps_flutter-0.5.21+7/ios/Classes/GoogleMapController.h:37:65: note: insert '_Nullable' if the pointer may be null
                        registrar:(NSObject<FlutterPluginRegistrar> *)registrar;
                                                                    ^
                                                                     _Nullable
    /Users/mozesong/Developer/flutter/.pub-cache/hosted/pub.dartlang.org/google_maps_flutter-0.5.21+7/ios/Classes/GoogleMapController.h:37:65: note: insert '_Nonnull' if the pointer should never be null
                        registrar:(NSObject<FlutterPluginRegistrar> *)registrar;
                                                                    ^
                                                                     _Nonnull
    /Users/mozesong/Developer/flutter/.pub-cache/hosted/pub.dartlang.org/google_maps_flutter-0.5.21+7/ios/Classes/GoogleMapController.h:40:50: warning: pointer is missing a nullability type specifier (_Nonnull, _Nullable, or _Null_unspecified) [-Wnullability-completeness]
    - (void)animateWithCameraUpdate:(GMSCameraUpdate *)cameraUpdate;
                                                     ^
    /Users/mozesong/Developer/flutter/.pub-cache/hosted/pub.dartlang.org/google_maps_flutter-0.5.21+7/ios/Classes/GoogleMapController.h:40:50: note: insert '_Nullable' if the pointer may be null
    - (void)animateWithCameraUpdate:(GMSCameraUpdate *)cameraUpdate;
                                                     ^
                                                      _Nullable
    /Users/mozesong/Developer/flutter/.pub-cache/hosted/pub.dartlang.org/google_maps_flutter-0.5.21+7/ios/Classes/GoogleMapController.h:40:50: note: insert '_Nonnull' if the pointer should never be null
    - (void)animateWithCameraUpdate:(GMSCameraUpdate *)cameraUpdate;
                                                     ^
                                                      _Nonnull
    /Users/mozesong/Developer/flutter/.pub-cache/hosted/pub.dartlang.org/google_maps_flutter-0.5.21+7/ios/Classes/GoogleMapController.h:41:47: warning: pointer is missing a nullability type specifier (_Nonnull, _Nullable, or _Null_unspecified) [-Wnullability-completeness]
    - (void)moveWithCameraUpdate:(GMSCameraUpdate *)cameraUpdate;
                                                  ^
    /Users/mozesong/Developer/flutter/.pub-cache/hosted/pub.dartlang.org/google_maps_flutter-0.5.21+7/ios/Classes/GoogleMapController.h:41:47: note: insert '_Nullable' if the pointer may be null
    - (void)moveWithCameraUpdate:(GMSCameraUpdate *)cameraUpdate;
                                                  ^
                                                   _Nullable
    /Users/mozesong/Developer/flutter/.pub-cache/hosted/pub.dartlang.org/google_maps_flutter-0.5.21+7/ios/Classes/GoogleMapController.h:41:47: note: insert '_Nonnull' if the pointer should never be null
    - (void)moveWithCameraUpdate:(GMSCameraUpdate *)cameraUpdate;
                                                  ^
                                                   _Nonnull
    /Users/mozesong/Developer/flutter/.pub-cache/hosted/pub.dartlang.org/google_maps_flutter-0.5.21+7/ios/Classes/GoogleMapController.h:42:22: warning: pointer is missing a nullability type specifier (_Nonnull, _Nullable, or _Null_unspecified) [-Wnullability-completeness]
    - (GMSCameraPosition *)cameraPosition;
                         ^
    /Users/mozesong/Developer/flutter/.pub-cache/hosted/pub.dartlang.org/google_maps_flutter-0.5.21+7/ios/Classes/GoogleMapController.h:42:22: note: insert '_Nullable' if the pointer may be null
    - (GMSCameraPosition *)cameraPosition;
                         ^
                          _Nullable
    /Users/mozesong/Developer/flutter/.pub-cache/hosted/pub.dartlang.org/google_maps_flutter-0.5.21+7/ios/Classes/GoogleMapController.h:42:22: note: insert '_Nonnull' if the pointer should never be null
    - (GMSCameraPosition *)cameraPosition;
                         ^
                          _Nonnull
    /Users/mozesong/Developer/flutter/.pub-cache/hosted/pub.dartlang.org/google_maps_flutter-0.5.21+7/ios/Classes/GoogleMapController.h:47:4: warning: pointer is missing a nullability type specifier (_Nonnull, _Nullable, or _Null_unspecified) [-Wnullability-completeness]
    - (instancetype)initWithRegistrar:(NSObject<FlutterPluginRegistrar> *)registrar;
       ^
    /Users/mozesong/Developer/flutter/.pub-cache/hosted/pub.dartlang.org/google_maps_flutter-0.5.21+7/ios/Classes/GoogleMapController.h:47:4: note: insert '_Nullable' if the pointer may be null
    - (instancetype)initWithRegistrar:(NSObject<FlutterPluginRegistrar> *)registrar;
       ^
                    _Nullable 
    /Users/mozesong/Developer/flutter/.pub-cache/hosted/pub.dartlang.org/google_maps_flutter-0.5.21+7/ios/Classes/GoogleMapController.h:47:4: note: insert '_Nonnull' if the pointer should never be null
    - (instancetype)initWithRegistrar:(NSObject<FlutterPluginRegistrar> *)registrar;
       ^
                    _Nonnull 
    /Users/mozesong/Developer/flutter/.pub-cache/hosted/pub.dartlang.org/google_maps_flutter-0.5.21+7/ios/Classes/GoogleMapController.h:47:69: warning: pointer is missing a nullability type specifier (_Nonnull, _Nullable, or _Null_unspecified) [-Wnullability-completeness]
    - (instancetype)initWithRegistrar:(NSObject<FlutterPluginRegistrar> *)registrar;
                                                                        ^
    /Users/mozesong/Developer/flutter/.pub-cache/hosted/pub.dartlang.org/google_maps_flutter-0.5.21+7/ios/Classes/GoogleMapController.h:47:69: note: insert '_Nullable' if the pointer may be null
    - (instancetype)initWithRegistrar:(NSObject<FlutterPluginRegistrar> *)registrar;
                                                                        ^
                                                                         _Nullable
    /Users/mozesong/Developer/flutter/.pub-cache/hosted/pub.dartlang.org/google_maps_flutter-0.5.21+7/ios/Classes/GoogleMapController.h:47:69: note: insert '_Nonnull' if the pointer should never be null
    - (instancetype)initWithRegistrar:(NSObject<FlutterPluginRegistrar> *)registrar;
                                                                        ^
                                                                         _Nonnull
    7 warnings generated.
    In file included from /Users/mozesong/Developer/flutter/.pub-cache/hosted/pub.dartlang.org/google_maps_flutter-0.5.21+7/ios/Classes/GoogleMapMarkerController.m:5:
    In file included from /Users/mozesong/Developer/flutter/.pub-cache/hosted/pub.dartlang.org/google_maps_flutter-0.5.21+7/ios/Classes/GoogleMapMarkerController.h:7:
    /Users/mozesong/Developer/flutter/.pub-cache/hosted/pub.dartlang.org/google_maps_flutter-0.5.21+7/ios/Classes/GoogleMapController.h:14:52: warning: pointer is missing a nullability type specifier (_Nonnull, _Nullable, or _Null_unspecified) [-Wnullability-completeness]
    - (void)setCameraTargetBounds:(GMSCoordinateBounds *)bounds;
                                                       ^
    /Users/mozesong/Developer/flutter/.pub-cache/hosted/pub.dartlang.org/google_maps_flutter-0.5.21+7/ios/Classes/GoogleMapController.h:14:52: note: insert '_Nullable' if the pointer may be null
    - (void)setCameraTargetBounds:(GMSCoordinateBounds *)bounds;
                                                       ^
                                                        _Nullable
    /Users/mozesong/Developer/flutter/.pub-cache/hosted/pub.dartlang.org/google_maps_flutter-0.5.21+7/ios/Classes/GoogleMapController.h:14:52: note: insert '_Nonnull' if the pointer should never be null
    - (void)setCameraTargetBounds:(GMSCoordinateBounds *)bounds;
                                                       ^
                                                        _Nonnull
    /Users/mozesong/Developer/flutter/.pub-cache/hosted/pub.dartlang.org/google_maps_flutter-0.5.21+7/ios/Classes/GoogleMapController.h:37:65: warning: pointer is missing a nullability type specifier (_Nonnull, _Nullable, or _Null_unspecified) [-Wnullability-completeness]
                        registrar:(NSObject<FlutterPluginRegistrar> *)registrar;
                                                                    ^
    /Users/mozesong/Developer/flutter/.pub-cache/hosted/pub.dartlang.org/google_maps_flutter-0.5.21+7/ios/Classes/GoogleMapController.h:37:65: note: insert '_Nullable' if the pointer may be null
                        registrar:(NSObject<FlutterPluginRegistrar> *)registrar;
                                                                    ^
                                                                     _Nullable
    /Users/mozesong/Developer/flutter/.pub-cache/hosted/pub.dartlang.org/google_maps_flutter-0.5.21+7/ios/Classes/GoogleMapController.h:37:65: note: insert '_Nonnull' if the pointer should never be null
                        registrar:(NSObject<FlutterPluginRegistrar> *)registrar;
                                                                    ^
                                                                     _Nonnull
    /Users/mozesong/Developer/flutter/.pub-cache/hosted/pub.dartlang.org/google_maps_flutter-0.5.21+7/ios/Classes/GoogleMapController.h:40:50: warning: pointer is missing a nullability type specifier (_Nonnull, _Nullable, or _Null_unspecified) [-Wnullability-completeness]
    - (void)animateWithCameraUpdate:(GMSCameraUpdate *)cameraUpdate;
                                                     ^
    /Users/mozesong/Developer/flutter/.pub-cache/hosted/pub.dartlang.org/google_maps_flutter-0.5.21+7/ios/Classes/GoogleMapController.h:40:50: note: insert '_Nullable' if the pointer may be null
    - (void)animateWithCameraUpdate:(GMSCameraUpdate *)cameraUpdate;
                                                     ^
                                                      _Nullable
    /Users/mozesong/Developer/flutter/.pub-cache/hosted/pub.dartlang.org/google_maps_flutter-0.5.21+7/ios/Classes/GoogleMapController.h:40:50: note: insert '_Nonnull' if the pointer should never be null
    - (void)animateWithCameraUpdate:(GMSCameraUpdate *)cameraUpdate;
                                                     ^
                                                      _Nonnull
    /Users/mozesong/Developer/flutter/.pub-cache/hosted/pub.dartlang.org/google_maps_flutter-0.5.21+7/ios/Classes/GoogleMapController.h:41:47: warning: pointer is missing a nullability type specifier (_Nonnull, _Nullable, or _Null_unspecified) [-Wnullability-completeness]
    - (void)moveWithCameraUpdate:(GMSCameraUpdate *)cameraUpdate;
                                                  ^
    /Users/mozesong/Developer/flutter/.pub-cache/hosted/pub.dartlang.org/google_maps_flutter-0.5.21+7/ios/Classes/GoogleMapController.h:41:47: note: insert '_Nullable' if the pointer may be null
    - (void)moveWithCameraUpdate:(GMSCameraUpdate *)cameraUpdate;
                                                  ^
                                                   _Nullable
    /Users/mozesong/Developer/flutter/.pub-cache/hosted/pub.dartlang.org/google_maps_flutter-0.5.21+7/ios/Classes/GoogleMapController.h:41:47: note: insert '_Nonnull' if the pointer should never be null
    - (void)moveWithCameraUpdate:(GMSCameraUpdate *)cameraUpdate;
                                                  ^
                                                   _Nonnull
    /Users/mozesong/Developer/flutter/.pub-cache/hosted/pub.dartlang.org/google_maps_flutter-0.5.21+7/ios/Classes/GoogleMapController.h:42:22: warning: pointer is missing a nullability type specifier (_Nonnull, _Nullable, or _Null_unspecified) [-Wnullability-completeness]
    - (GMSCameraPosition *)cameraPosition;
                         ^
    /Users/mozesong/Developer/flutter/.pub-cache/hosted/pub.dartlang.org/google_maps_flutter-0.5.21+7/ios/Classes/GoogleMapController.h:42:22: note: insert '_Nullable' if the pointer may be null
    - (GMSCameraPosition *)cameraPosition;
                         ^
                          _Nullable
    /Users/mozesong/Developer/flutter/.pub-cache/hosted/pub.dartlang.org/google_maps_flutter-0.5.21+7/ios/Classes/GoogleMapController.h:42:22: note: insert '_Nonnull' if the pointer should never be null
    - (GMSCameraPosition *)cameraPosition;
                         ^
                          _Nonnull
    /Users/mozesong/Developer/flutter/.pub-cache/hosted/pub.dartlang.org/google_maps_flutter-0.5.21+7/ios/Classes/GoogleMapController.h:47:4: warning: pointer is missing a nullability type specifier (_Nonnull, _Nullable, or _Null_unspecified) [-Wnullability-completeness]
    - (instancetype)initWithRegistrar:(NSObject<FlutterPluginRegistrar> *)registrar;
       ^
    /Users/mozesong/Developer/flutter/.pub-cache/hosted/pub.dartlang.org/google_maps_flutter-0.5.21+7/ios/Classes/GoogleMapController.h:47:4: note: insert '_Nullable' if the pointer may be null
    - (instancetype)initWithRegistrar:(NSObject<FlutterPluginRegistrar> *)registrar;
       ^
                    _Nullable 
    /Users/mozesong/Developer/flutter/.pub-cache/hosted/pub.dartlang.org/google_maps_flutter-0.5.21+7/ios/Classes/GoogleMapController.h:47:4: note: insert '_Nonnull' if the pointer should never be null
    - (instancetype)initWithRegistrar:(NSObject<FlutterPluginRegistrar> *)registrar;
       ^
                    _Nonnull 
    /Users/mozesong/Developer/flutter/.pub-cache/hosted/pub.dartlang.org/google_maps_flutter-0.5.21+7/ios/Classes/GoogleMapController.h:47:69: warning: pointer is missing a nullability type specifier (_Nonnull, _Nullable, or _Null_unspecified) [-Wnullability-completeness]
    - (instancetype)initWithRegistrar:(NSObject<FlutterPluginRegistrar> *)registrar;
                                                                        ^
    /Users/mozesong/Developer/flutter/.pub-cache/hosted/pub.dartlang.org/google_maps_flutter-0.5.21+7/ios/Classes/GoogleMapController.h:47:69: note: insert '_Nullable' if the pointer may be null
    - (instancetype)initWithRegistrar:(NSObject<FlutterPluginRegistrar> *)registrar;
                                                                        ^
                                                                         _Nullable
    /Users/mozesong/Developer/flutter/.pub-cache/hosted/pub.dartlang.org/google_maps_flutter-0.5.21+7/ios/Classes/GoogleMapController.h:47:69: note: insert '_Nonnull' if the pointer should never be null
    - (instancetype)initWithRegistrar:(NSObject<FlutterPluginRegistrar> *)registrar;
                                                                        ^
                                                                         _Nonnull
    7 warnings generated.
    In file included from /Users/mozesong/Developer/flutter/.pub-cache/hosted/pub.dartlang.org/google_maps_flutter-0.5.21+7/ios/Classes/GoogleMapController.m:5:
    /Users/mozesong/Developer/flutter/.pub-cache/hosted/pub.dartlang.org/google_maps_flutter-0.5.21+7/ios/Classes/GoogleMapController.h:14:52: warning: pointer is missing a nullability type specifier (_Nonnull, _Nullable, or _Null_unspecified) [-Wnullability-completeness]
    - (void)setCameraTargetBounds:(GMSCoordinateBounds *)bounds;
                                                       ^
    /Users/mozesong/Developer/flutter/.pub-cache/hosted/pub.dartlang.org/google_maps_flutter-0.5.21+7/ios/Classes/GoogleMapController.h:14:52: note: insert '_Nullable' if the pointer may be null
    - (void)setCameraTargetBounds:(GMSCoordinateBounds *)bounds;
                                                       ^
                                                        _Nullable
    /Users/mozesong/Developer/flutter/.pub-cache/hosted/pub.dartlang.org/google_maps_flutter-0.5.21+7/ios/Classes/GoogleMapController.h:14:52: note: insert '_Nonnull' if the pointer should never be null
    - (void)setCameraTargetBounds:(GMSCoordinateBounds *)bounds;
                                                       ^
                                                        _Nonnull
    /Users/mozesong/Developer/flutter/.pub-cache/hosted/pub.dartlang.org/google_maps_flutter-0.5.21+7/ios/Classes/GoogleMapController.h:37:65: warning: pointer is missing a nullability type specifier (_Nonnull, _Nullable, or _Null_unspecified) [-Wnullability-completeness]
                        registrar:(NSObject<FlutterPluginRegistrar> *)registrar;
                                                                    ^
    /Users/mozesong/Developer/flutter/.pub-cache/hosted/pub.dartlang.org/google_maps_flutter-0.5.21+7/ios/Classes/GoogleMapController.h:37:65: note: insert '_Nullable' if the pointer may be null
                        registrar:(NSObject<FlutterPluginRegistrar> *)registrar;
                                                                    ^
                                                                     _Nullable
    /Users/mozesong/Developer/flutter/.pub-cache/hosted/pub.dartlang.org/google_maps_flutter-0.5.21+7/ios/Classes/GoogleMapController.h:37:65: note: insert '_Nonnull' if the pointer should never be null
                        registrar:(NSObject<FlutterPluginRegistrar> *)registrar;
                                                                    ^
                                                                     _Nonnull
    /Users/mozesong/Developer/flutter/.pub-cache/hosted/pub.dartlang.org/google_maps_flutter-0.5.21+7/ios/Classes/GoogleMapController.h:40:50: warning: pointer is missing a nullability type specifier (_Nonnull, _Nullable, or _Null_unspecified) [-Wnullability-completeness]
    - (void)animateWithCameraUpdate:(GMSCameraUpdate *)cameraUpdate;
                                                     ^
    /Users/mozesong/Developer/flutter/.pub-cache/hosted/pub.dartlang.org/google_maps_flutter-0.5.21+7/ios/Classes/GoogleMapController.h:40:50: note: insert '_Nullable' if the pointer may be null
    - (void)animateWithCameraUpdate:(GMSCameraUpdate *)cameraUpdate;
                                                     ^
                                                      _Nullable
    /Users/mozesong/Developer/flutter/.pub-cache/hosted/pub.dartlang.org/google_maps_flutter-0.5.21+7/ios/Classes/GoogleMapController.h:40:50: note: insert '_Nonnull' if the pointer should never be null
    - (void)animateWithCameraUpdate:(GMSCameraUpdate *)cameraUpdate;
                                                     ^
                                                      _Nonnull
    /Users/mozesong/Developer/flutter/.pub-cache/hosted/pub.dartlang.org/google_maps_flutter-0.5.21+7/ios/Classes/GoogleMapController.h:41:47: warning: pointer is missing a nullability type specifier (_Nonnull, _Nullable, or _Null_unspecified) [-Wnullability-completeness]
    - (void)moveWithCameraUpdate:(GMSCameraUpdate *)cameraUpdate;
                                                  ^
    /Users/mozesong/Developer/flutter/.pub-cache/hosted/pub.dartlang.org/google_maps_flutter-0.5.21+7/ios/Classes/GoogleMapController.h:41:47: note: insert '_Nullable' if the pointer may be null
    - (void)moveWithCameraUpdate:(GMSCameraUpdate *)cameraUpdate;
                                                  ^
                                                   _Nullable
    /Users/mozesong/Developer/flutter/.pub-cache/hosted/pub.dartlang.org/google_maps_flutter-0.5.21+7/ios/Classes/GoogleMapController.h:41:47: note: insert '_Nonnull' if the pointer should never be null
    - (void)moveWithCameraUpdate:(GMSCameraUpdate *)cameraUpdate;
                                                  ^
                                                   _Nonnull
    /Users/mozesong/Developer/flutter/.pub-cache/hosted/pub.dartlang.org/google_maps_flutter-0.5.21+7/ios/Classes/GoogleMapController.h:42:22: warning: pointer is missing a nullability type specifier (_Nonnull, _Nullable, or _Null_unspecified) [-Wnullability-completeness]
    - (GMSCameraPosition *)cameraPosition;
                         ^
    /Users/mozesong/Developer/flutter/.pub-cache/hosted/pub.dartlang.org/google_maps_flutter-0.5.21+7/ios/Classes/GoogleMapController.h:42:22: note: insert '_Nullable' if the pointer may be null
    - (GMSCameraPosition *)cameraPosition;
                         ^
                          _Nullable
    /Users/mozesong/Developer/flutter/.pub-cache/hosted/pub.dartlang.org/google_maps_flutter-0.5.21+7/ios/Classes/GoogleMapController.h:42:22: note: insert '_Nonnull' if the pointer should never be null
    - (GMSCameraPosition *)cameraPosition;
                         ^
                          _Nonnull
    /Users/mozesong/Developer/flutter/.pub-cache/hosted/pub.dartlang.org/google_maps_flutter-0.5.21+7/ios/Classes/GoogleMapController.h:47:4: warning: pointer is missing a nullability type specifier (_Nonnull, _Nullable, or _Null_unspecified) [-Wnullability-completeness]
    - (instancetype)initWithRegistrar:(NSObject<FlutterPluginRegistrar> *)registrar;
       ^
    /Users/mozesong/Developer/flutter/.pub-cache/hosted/pub.dartlang.org/google_maps_flutter-0.5.21+7/ios/Classes/GoogleMapController.h:47:4: note: insert '_Nullable' if the pointer may be null
    - (instancetype)initWithRegistrar:(NSObject<FlutterPluginRegistrar> *)registrar;
       ^
                    _Nullable 
    /Users/mozesong/Developer/flutter/.pub-cache/hosted/pub.dartlang.org/google_maps_flutter-0.5.21+7/ios/Classes/GoogleMapController.h:47:4: note: insert '_Nonnull' if the pointer should never be null
    - (instancetype)initWithRegistrar:(NSObject<FlutterPluginRegistrar> *)registrar;
       ^
                    _Nonnull 
    /Users/mozesong/Developer/flutter/.pub-cache/hosted/pub.dartlang.org/google_maps_flutter-0.5.21+7/ios/Classes/GoogleMapController.h:47:69: warning: pointer is missing a nullability type specifier (_Nonnull, _Nullable, or _Null_unspecified) [-Wnullability-completeness]
    - (instancetype)initWithRegistrar:(NSObject<FlutterPluginRegistrar> *)registrar;
                                                                        ^
    /Users/mozesong/Developer/flutter/.pub-cache/hosted/pub.dartlang.org/google_maps_flutter-0.5.21+7/ios/Classes/GoogleMapController.h:47:69: note: insert '_Nullable' if the pointer may be null
    - (instancetype)initWithRegistrar:(NSObject<FlutterPluginRegistrar> *)registrar;
                                                                        ^
                                                                         _Nullable
    /Users/mozesong/Developer/flutter/.pub-cache/hosted/pub.dartlang.org/google_maps_flutter-0.5.21+7/ios/Classes/GoogleMapController.h:47:69: note: insert '_Nonnull' if the pointer should never be null
    - (instancetype)initWithRegistrar:(NSObject<FlutterPluginRegistrar> *)registrar;
                                                                        ^
                                                                         _Nonnull
    7 warnings generated.
    /Users/mozesong/Developer/flutter/.pub-cache/hosted/pub.dartlang.org/google_sign_in-4.0.7/ios/Classes/GoogleSignInPlugin.m:137:41: warning: 'UIApplicationOpenURLOptionsSourceApplicationKey' is only available on iOS 9.0 or newer [-Wunguarded-availability]
      NSString *sourceApplication = options[UIApplicationOpenURLOptionsSourceApplicationKey];
                                            ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    In module 'UIKit' imported from /Users/mozesong/Documents/GitHub/Green-Hero-Apps-2.0/green_hero_vendor/ios/Pods/Target Support Files/google_sign_in/google_sign_in-prefix.pch:2:
    /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator12.4.sdk/System/Library/Frameworks/UIKit.framework/Headers/UIApplication.h:538:51: note: 'UIApplicationOpenURLOptionsSourceApplicationKey' has been marked as being introduced in iOS 9.0 here, but the deployment target is iOS 8.0.0
    UIKIT_EXTERN UIApplicationOpenURLOptionsKey const UIApplicationOpenURLOptionsSourceApplicationKey NS_AVAILABLE_IOS(9_0);   // value is an NSString containing the bundle ID of the originating application
                                                      ^
    /Users/mozesong/Developer/flutter/.pub-cache/hosted/pub.dartlang.org/google_sign_in-4.0.7/ios/Classes/GoogleSignInPlugin.m:137:41: note: enclose 'UIApplicationOpenURLOptionsSourceApplicationKey' in an @available check to silence this warning
      NSString *sourceApplication = options[UIApplicationOpenURLOptionsSourceApplicationKey];
                                            ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    /Users/mozesong/Developer/flutter/.pub-cache/hosted/pub.dartlang.org/google_sign_in-4.0.7/ios/Classes/GoogleSignInPlugin.m:138:27: warning: 'UIApplicationOpenURLOptionsAnnotationKey' is only available on iOS 9.0 or newer [-Wunguarded-availability]
      id annotation = options[UIApplicationOpenURLOptionsAnnotationKey];
                              ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    In module 'UIKit' imported from /Users/mozesong/Documents/GitHub/Green-Hero-Apps-2.0/green_hero_vendor/ios/Pods/Target Support Files/google_sign_in/google_sign_in-prefix.pch:2:
    /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator12.4.sdk/System/Library/Frameworks/UIKit.framework/Headers/UIApplication.h:539:51: note: 'UIApplicationOpenURLOptionsAnnotationKey' has been marked as being introduced in iOS 9.0 here, but the deployment target is iOS 8.0.0
    UIKIT_EXTERN UIApplicationOpenURLOptionsKey const UIApplicationOpenURLOptionsAnnotationKey NS_AVAILABLE_IOS(9_0);   // value is a property-list typed object corresponding to what the originating application passed in UIDocumentInteractionController's annotation property
                                                      ^
    /Users/mozesong/Developer/flutter/.pub-cache/hosted/pub.dartlang.org/google_sign_in-4.0.7/ios/Classes/GoogleSignInPlugin.m:138:27: note: enclose 'UIApplicationOpenURLOptionsAnnotationKey' in an @available check to silence this warning
      id annotation = options[UIApplicationOpenURLOptionsAnnotationKey];
                              ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    2 warnings generated.
    error: /Users/mozesong/Developer/flutter/.pub-cache/hosted/pub.dartlang.org/firebase_auth-0.14.0+5/ios/Classes/UserAgent.h: No such file or directory
    error: /Users/mozesong/Developer/flutter/.pub-cache/hosted/pub.dartlang.org/firebase_storage-3.0.6/ios/Classes/UserAgent.h: No such file or directory
    error: /Users/mozesong/Developer/flutter/.pub-cache/hosted/pub.dartlang.org/firebase_core-0.4.0+9/ios/Classes/UserAgent.h: No such file or directory
    Runner-hkczeqbdyzwqvrazjtxoyzhjoorv
    note: Using new build systemnote: Planning buildnote: Constructing build description
```

## Flutter doctor:
```
[✓] Flutter (Channel stable, v1.9.1+hotfix.2, on Mac OS X 10.14.6 18G103, locale en-MY)
 
[✓] Android toolchain - develop for Android devices (Android SDK version 29.0.2)
[✓] Xcode - develop for iOS and macOS (Xcode 10.3)
[✓] Android Studio (version 3.4)
[✓] Connected device (2 available)

• No issues found!
```

**Comment by kedarprabhu46** [comment]
Changing the deployment target for the Runner project in Xcode to iOS 10.0 resolved this error for me.

**Comment by momoDragon** [comment]
> Changing the deployment target for the Runner project in Xcode to iOS 10.0 resolved this error for me.

is it in podfile?
`platform :ios, '10.0'
`
The same error was given. Is there any fix yet? @janmoppel

**Comment by momoDragon** [comment]
Hey guys any fix yet?

**Comment by kedarprabhu46** [comment]
> > Changing the deployment target for the Runner project in Xcode to iOS 10.0 resolved this error for me.
> 
> is it in podfile?
> `platform :ios, '10.0' `
> The same error was given. Is there any fix yet? @janmoppel

No you have to change it from xcode under target platform

**Comment by logemann** [comment]
btw. Got the same error after adding google_maps_flutter . There should be a hint somewhere on the plugin page.

**Comment by basemelyyan** [comment]
I got the same error.

**Comment by jmagman** [comment]
Ignoring the `@available` warnings, the actual errors are:

```
 error: /Users/mozesong/Developer/flutter/.pub-cache/hosted/pub.dartlang.org/firebase_auth-0.14.0+5/ios/Classes/UserAgent.h: No such file or directory
error: /Users/mozesong/Developer/flutter/.pub-cache/hosted/pub.dartlang.org/firebase_storage-3.0.6/ios/Classes/UserAgent.h: No such file or directory
error: /Users/mozesong/Developer/flutter/.pub-cache/hosted/pub.dartlang.org/firebase_core-0.4.0+9/ios/Classes/UserAgent.h: No such file or directory
```
Those missing files should be coming from the [firebase* podspec `prepare_command`](https://github.com/FirebaseExtended/flutterfire/blob/7153c28f7517518157008e5410057205c6420860/packages/firebase_dynamic_links/ios/firebase_dynamic_links.podspec#L28) which only get run when the pod is first downloaded.  I don't know how this works when the pod isn't being downloaded via `pod` but instead via a symlink to the pub-cache...
I was able to reproduce this by adding this to a pubspec:
```
  firebase_auth: 
  firebase_storage: 
  firebase_core: 
```
Running `flutter pub get` then running:
```
flutter pub cache repair
```
which deleted the UserAgent.h files.

I also don't know why adding `google_maps_flutter` would make this start happening since AFAIK it doesn't have any firebase dependencies.  I'm guessing the user was really seeing some other issue, and in an attempt to fix it by repairing the cache, hit this other issue where the UserAgent header disappears and doesn't get regenerated.

Duplicate of https://github.com/FirebaseExtended/flutterfire/issues/1730 but that one was closed for  even though this is still an issue.

Essentially: `s.prepare_command` shouldn't be relied on in a podspec that is distributed via pub.
